lms/djangoapps/certificates/utils.py

```diff
@@ -250,3 +256,24 @@ def get_preferred_certificate_name(user):
         name_to_use = ''
 
     return name_to_use
+
+
+def get_certificate_configuration_from_credentials(course_id: str, mode: str) -> Optional[Response]:
+    """
+    Makes a request to the credentials service to get a certificate configuration of the course.
+    """
+    try:
+        credentials_client = get_credentials_api_client(
+            User.objects.get(username=settings.CREDENTIALS_SERVICE_USERNAME)
+        )
+    except User.DoesNotExist:
+        log.warning(f'Credentials service user [{settings.CREDENTIALS_SERVICE_USERNAME}] does not exist')
```

Comment:
Please describe why we need this and why we can return nothing?

Reply:
Because in this way you can find out why it is not possible to make a request to Credentials. I had a case on another project, when it was precisely because of the absence of this user that certificates for programs were not generated and there were no logs about this.

Reply:
Refactored, according to an already existing implementation https://github.com/raccoongang/edx-platform/blob/master/openedx/core/djangoapps/credentials/utils.py#L112